README.md

```diff
@@ -1,160 +1,718 @@
 
 ---
 
-## Getting started 🐣
+## About Search UI :rocket:
+
+Search UI allows you to quickly implement search experiences without re-inventing the wheel.
+
+It supports **[React](https://reactjs.org)** and **works with any search API**.
 
-Search UI supports any search engine provider and JavaScript framework.
+### Features :+1:
 
-It is developed and maintained by [Elastic](https://elastic.co).
+- **You know, for search**: Maintained by [Elastic](https://elastic.co), the team behind Elasticsearch.
+- **Speedy Implementation** - Build search with a search box, results view, sorting, and more, with a few lines of code.
+- **Customizable** - Tune the Components, markup, styles, and behaviors to your liking.
+- **Smart URLs** - Searches, paging, filtering, and more, are captured in the URL for direct result linking.
+- **Headless** - Leverage our application logic, provide your own Components or views.
+- **JavaScript-y** - Can be used used with any JavaScript library, even vanilla JavaScript.
+
+<img src="packages/react-search-ui/resources/screenshot.png" width="600">
+
+## Getting started 🐣
 
-You can use Search UI to quickly build robust, enjoyable, fully-featured search experiences.
+**Looking for a great search API?** [Elastic App Search](https://www.elastic.co/cloud/app-search-service) has a slick dashboard, powerful features, and leading relevance.
 
-To begin, choose your framework:
+Install the **React Search UI** and the **App Search** connector.
 
-- [React](packages/react-search-ui/README.md)
-- More coming! [Contributions](#contribute-) welcome.
+```sh
+# Install React Search UI
+npm install --save @elastic/react-search-ui
+
+# Install a Connector, like the Elastic's App Search Connector
+# You can use a custom connector to use any search API.
+npm install --save  @elastic/search-ui-app-search-connector
+```
 
 _Note: The Search UI is in beta. We do not recommend production use._
 
-## Developing Search UI
+## Creating a Search UI
+
+<a id="search-ui"></a>
+
+Use out of the box components, styles, and layouts to build a search experience quickly.
+
+```jsx
+import React from "react";
+import AppSearchAPIConnector from "@elastic/search-ui-app-search-connector";
+import { SearchProvider, Results, SearchBox } from "@elastic/react-search-ui";
+import { Layout } from "@elastic/react-search-ui-views";
+
+import "@elastic/react-search-ui-views/lib/styles/styles.css";
+
+const connector = new AppSearchAPIConnector({
+  searchKey: "search-371auk61r2bwqtdzocdgutmg",
+  engineName: "search-ui-examples",
+  hostIdentifier: "host-2376rb"
+});
+
+export default function App() {
+  return (
+    <SearchProvider
+      config={{
+        apiConnector: connector
+      }}
+    >
+      {() => (
+        <div className="App">
+          <Layout
+            header={<SearchBox />}
+            bodyContent={<Results titleField="title" urlField="nps_link" />}
+          />
+        </div>
+      )}
+    </SearchProvider>
+  );
+}
+```
```

Comment:
I think another key part of this library that we need to highlight in this README is the *headless* part of this. Something that very simply showcases this library in the same way as https://github.com/downshift-js/downshift and https://github.com/jaredpalmer/formik.

We could have two examples in this section:

```
Use out of the box components, styles, and layouts to build a search experience quickly.

<existing example here>

Or go headless

<SearchProvider config={config}>
  {{ searchTerm, setSearchTerm, results } => {
    return (
      <div>
        <input value={searchTerm} onChange={setSearchTerm} />
         { results.map( r => (
            <div>{r.title}</div>
         ) ) }
      </div>
    );
  }}
</SearchProvider>
```

I could give you a better example, that one is pseudo code.

Reply:
@JasonStoltz Nice -- I like the pseudo example.